project/SparkBuild.scala

```diff
@@ -802,11 +805,29 @@ object Hive {
 }
 
 object YARN {
+  val genConfigProperties = TaskKey[Unit]("gen-config-properties",
+    "Generate config.properties which contains a setting whether Hadoop is provided or not")
+  val propFileName = "config.properties"
+
   lazy val settings = Seq(
     excludeDependencies --= Seq(
       ExclusionRule(organization = "com.sun.jersey"),
       ExclusionRule("javax.servlet", "javax.servlet-api"),
-      ExclusionRule("javax.ws.rs", "jsr311-api"))
+      ExclusionRule("javax.ws.rs", "jsr311-api")),
+      Compile / unmanagedResources :=
+        (Compile / unmanagedResources).value.filter(!_.getName.endsWith(s"$propFileName")),
+      genConfigProperties := {
+        val file = (Compile / classDirectory).value / s"org/apache/spark/deploy/yarn/$propFileName"
+        val isHadoopProvided = sys.props.getOrElse("spark.yarn.isHadoopProvided", "false")
+        IO.write(file, s"spark.yarn.isHadoopProvided = $isHadoopProvided")
+      },
+      (Compile / copyResources) := (Def.taskDyn {
+        val c = (Compile / copyResources).value
+          Def.task {
```

Comment:
looks like we should indent 2 spaces back here too

Reply:
Oh, I was confused because `Seq` is nested... Thanks.